ports/geckolib/glue.rs

```diff
@@ -2263,9 +2277,14 @@ macro_rules! get_property_id_from_property {
 
 fn get_property_value(declarations: RawServoDeclarationBlockBorrowed,
                       property_id: PropertyId, value: *mut nsAString) {
-    read_locked_arc(declarations, |decls: &PropertyDeclarationBlock| {
-        decls.property_value_to_css(&property_id, unsafe { value.as_mut().unwrap() }).unwrap();
-    })
+    // This callsite is hot enough that the lock acquisition shows up in profiles.
+    // Using an unchecked read here improves our performance by ~10% on the
+    // microbenchmark in bug 1355599.
```

Comment:
How is this not unsound? That should clearly be explained.

Reply:
Because there's only one SharedLock in Gecko, and it's backed by an AtomicRefCell, and thus would panic if the invariants were ever violated (this is all gecko-specific FFI code). So this is all just belt-and-suspenders, and less relevant to Gecko which binds all CSSOM stuff to the main thread.

Reply:
What would panic?

Reply:
Acquiring the SharedLock, in the case that it was already acquired, would panic, rather than block (since it's backed by an AtomicRefCell in Gecko, as opposed to an RwLock in Servo). This is why this machinery is just a release-mode assertion in geckolib builds, and the condition it's asserting against is not particularly likely.